ee/localserver/zta_auth_middleware_test.go

```diff
@@ -0,0 +1,336 @@
+
+	handler := ztaMiddleware.Wrap(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write(returnData)
+		w.WriteHeader(http.StatusOK)
+	}))
+
+	t.Run("handles missing box param", func(t *testing.T) {
+		t.Parallel()
+		rr := httptest.NewRecorder()
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, "/", nil))
+		require.Equal(t, http.StatusBadRequest, rr.Code,
+			"should return bad request when box param is missing",
+		)
+	})
+
+	t.Run("handles bad b64", func(t *testing.T) {
+		t.Parallel()
+		rr := httptest.NewRecorder()
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, "/?payload=badb64", nil))
+		require.Equal(t, http.StatusBadRequest, rr.Code,
+			"should return bad request when box param is not valid b64",
+		)
+	})
+
+	t.Run("handles chain unmarshall failure", func(t *testing.T) {
+		t.Parallel()
+		rr := httptest.NewRecorder()
+		encoded := base64.URLEncoding.EncodeToString([]byte("badchain"))
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, fmt.Sprintf("/?payload=%s", encoded), nil))
+		require.Equal(t, http.StatusBadRequest, rr.Code,
+			"should return bad request when chain cannot be unmarshalled",
+		)
+	})
+
+	t.Run("handles invalid chain", func(t *testing.T) {
+		t.Parallel()
+
+		invalidKeys := make([]*ecdsa.PrivateKey, 3)
+		for i := range invalidKeys {
+			invalidKeys[i] = mustGenEcdsaKey(t)
+		}
+
+		pubcallerPubKey, _, err := box.GenerateKey(rand.Reader)
+		require.NoError(t, err)
+
+		chain, err := newChain(pubcallerPubKey, invalidKeys...)
+		require.NoError(t, err)
+
+		chainMarshalled, err := json.Marshal(chain)
+		require.NoError(t, err)
+
+		b64 := base64.URLEncoding.EncodeToString(chainMarshalled)
+
+		rr := httptest.NewRecorder()
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, fmt.Sprintf("/?payload=%s", url.QueryEscape(b64)), nil))
+		require.Equal(t, http.StatusUnauthorized, rr.Code,
+			"should return unauthorized when chain cannot be validated",
+		)
+	})
+
+	t.Run("handles valid chain", func(t *testing.T) {
+		t.Parallel()
+
+		validKeys := make([]*ecdsa.PrivateKey, 4)
+		validKeys[0] = rootTrustedEcKey
+
+		for i := 1; i < len(validKeys); i++ {
+			validKeys[i] = mustGenEcdsaKey(t)
+		}
+
+		callerPubKey, callerPrivKey, err := box.GenerateKey(rand.Reader)
+		require.NoError(t, err)
+
+		chain, err := newChain(callerPubKey, validKeys...)
+		require.NoError(t, err)
+
+		chainMarshalled, err := json.Marshal(chain)
+		require.NoError(t, err)
+
+		b64 := base64.URLEncoding.EncodeToString(chainMarshalled)
+
+		rr := httptest.NewRecorder()
+		handler.ServeHTTP(rr, httptest.NewRequest(http.MethodGet, fmt.Sprintf("/?payload=%s", url.QueryEscape(b64)), nil))
+		require.Equal(t, http.StatusOK, rr.Code,
+			"should return ok when chain is valid",
+		)
+
+		bodyBytes, err := io.ReadAll(rr.Body)
+		require.NoError(t, err)
+
+		bodyDecoded, err := base64.StdEncoding.DecodeString(string(bodyBytes))
+		require.NoError(t, err)
+
+		var ztaResponse ztaResponse
```

Comment:
Nitpick -- maybe `var z ztaResponse` or `var resp ztaResponse` to avoid shadowing?